File: lua/triangle/triangle.lua

local classify_triangle = function (a, b, c)
    local sides = {a, b, c}
    table.sort(sides)
    a, b, c = table.unpack(sides)

    assert(a > 0 and a + b > c, 'Input Error')

    if a == b and a == c then
        return 'equilateral'
    elseif a == b or a == c or b == c then
        return 'isosceles'
    else
        return 'scalene'
    end
end

return { kind = classify_triangle }

File: swift/triangle/Sources/Triangle/Triangle.swift

struct Triangle {
    private(set) var kind: String

    init(_ a: Double, _ b: Double, _ c: Double) {
        let sides = [a, b, c].sorted()
        if sides[0] <= 0 || sides[0] + sides[1] <= sides[2] {
            kind = "ErrorKind"
        } else {
            let s: Set = [a, b, c]
            switch s.count {
                case 1:  kind = "Equilateral"
                case 2:  kind = "Isosceles"
                default: kind = "Scalene"
            }
        }
    }
}
